srs/src/voice_stream.rs

```diff
@@ -0,0 +1,356 @@
+        let s = self.get_mut();
+        s.voice_sink.as_mut().poll_ready(cx)
+    }
+
+    fn start_send(self: Pin<&mut Self>, item: Vec<u8>) -> Result<(), Self::Error> {
+        let mut sguid = [0; 22];
+        sguid.clone_from_slice(self.client.sguid().as_bytes());
+
+        let packet = VoicePacket {
+            audio_part: item,
+            frequencies: vec![Frequency {
+                freq: self.client.freq() as f64,
+                modulation: if self.client.freq() <= 87_995_000 {
+                    Modulation::Fm
+                } else {
+                    Modulation::Am
+                },
+                encryption: Encryption::None,
+            }],
+            unit_id: self.client.unit().map(|u| u.id).unwrap_or(0),
+            packet_id: self.packet_id,
+            hop_count: 0,
+            transmission_sguid: sguid,
+            client_sguid: sguid,
+        };
+
+        let s = self.get_mut();
+        s.packet_id = s.packet_id.wrapping_add(1);
+
+        s.voice_sink.as_mut().start_send(packet.into())
+    }
+
+    fn poll_flush(self: Pin<&mut Self>, cx: &mut Context) -> Poll<Result<(), Self::Error>> {
+        let s = self.get_mut();
+        s.voice_sink.as_mut().poll_flush(cx)
+    }
+
+    fn poll_close(self: Pin<&mut Self>, cx: &mut Context) -> Poll<Result<(), Self::Error>> {
+        let s = self.get_mut();
+        s.voice_sink.as_mut().poll_close(cx)
+    }
+}
+
+impl Sink<VoicePacket> for VoiceStream {
+    type Error = PollSendError<Packet>;
+
+    fn poll_ready(self: Pin<&mut Self>, cx: &mut Context) -> Poll<Result<(), Self::Error>> {
+        let s = self.get_mut();
+        s.voice_sink.as_mut().poll_ready(cx)
+    }
+
+    fn start_send(self: Pin<&mut Self>, mut packet: VoicePacket) -> Result<(), Self::Error> {
+        let mut sguid = [0; 22];
+        sguid.clone_from_slice(self.client.sguid().as_bytes());
+        packet.client_sguid = sguid;
+
+        let s = self.get_mut();
+        s.packet_id = s.packet_id.wrapping_add(1);
+
+        s.voice_sink.as_mut().start_send(packet.into())
+    }
+
+    fn poll_flush(self: Pin<&mut Self>, cx: &mut Context) -> Poll<Result<(), Self::Error>> {
+        let s = self.get_mut();
+        s.voice_sink.as_mut().poll_flush(cx)
+    }
+
+    fn poll_close(self: Pin<&mut Self>, cx: &mut Context) -> Poll<Result<(), Self::Error>> {
+        let s = self.get_mut();
+        s.voice_sink.as_mut().poll_close(cx)
+    }
+}
+
+async fn create_radio_update_message(client: &Client) -> Message {
+    let pos = client.position().await;
+    Message {
+        client: Some(MsgClient {
+            client_guid: client.sguid().to_string(),
+            name: Some(client.name().to_string()),
+            coalition: client.coalition,
+            radio_info: Some(RadioInfo {
+                name: "DATIS Radios".to_string(),
+                ptt: false,
+                // TODO: enable one of the radios to receive voice
```

Comment:
Is this a TODO to act as a workaround to some issue or to porentially open up the idea of a STT mechanism?